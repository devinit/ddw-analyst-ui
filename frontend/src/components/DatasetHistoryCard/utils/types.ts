export interface SavedQueryData {
  id: number;
  saved_query_db_table: string;
  operation: number;
  full_query: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
}
