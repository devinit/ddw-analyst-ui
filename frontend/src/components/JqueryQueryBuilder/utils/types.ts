export type JqueryQueryBuilderFilter = {
  condition?: string;
  rules?: (JqueryQueryBuilderFilterComparator | JqueryQueryBuilderFilter)[];
};

export type JqueryQueryBuilderHaving = {
  condition?: string;
  rules?: (JqueryQueryBuilderHavingComparator | JqueryQueryBuilderHaving)[];
};

export type JqueryQueryBuilderHavingComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value?: string | number | { plain: string | number } | { column: string; aggregate: string };
  values?: { value: string; label: string }[];
};

export type JqueryQueryBuilderFilterComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value: string | number;
};
export type JqueryQueryBuilderComps =
  | 'equal'
  | 'not_equal'
  | 'less'
  | 'less_or_equal'
  | 'greater'
  | 'greater_or_equal'
  | 'between'
  | 'in';

export type JqueryQueryBuilderFieldData = {
  id: string | number;
  label: string;
  type: string;
};

export type JqueryQueryBuilderIcons = {
  add_group: string;
  add_rule: string;
  remove_group: string;
  remove_rule: string;
  error: string;
};
