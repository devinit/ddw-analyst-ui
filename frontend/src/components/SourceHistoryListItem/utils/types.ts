export interface FrozenData {
  id: number;
  parent_db_table: string;
  frozen_db_table: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
  logs: string | null;
}
