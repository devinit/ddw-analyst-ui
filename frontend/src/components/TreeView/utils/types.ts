import { ReactElement } from 'react';

export interface TreeViewProps {
  data: Data[];
  depth?: number;

  deleteElement?: ReactElement;

  getStyleClassCb?: (node: Data) => void;

  isCheckable?: (node: Data, depth?: number) => void;
  isDeletable?: (node: Data, depth?: number) => void;
  isExpandable?: (node: Data, depth?: number) => void;

  keywordChildren?: string;
  keywordChildrenLoading?: string;
  keywordKey?: string;
  keywordLabel?: string;

  loadingElement?: ReactElement;
  noChildrenAvailableMessage?: string;

  onCheckToggleCb?: (data: Data[], depth?: number) => void;
  onDeleteCb?: (node: Data, data: Data[], depth?: number) => void;
  onExpandToggleCb?: (node: Data, depth?: number) => void;
  onUpdateCb?: (data: unknown[], depth?: number) => void;

  transitionEnterTimeout?: number;
  transitionExitTimeout?: number;
}

export interface Data {
  id: number;
  name: string;
  isExpanded?: boolean;
  isChecked?: boolean;
  isChildrenLoading?: boolean;
  children?: Data[];
}
