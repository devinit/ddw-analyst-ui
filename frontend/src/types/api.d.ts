export interface APIResponse<T> {
  count: number;
  next: string;
  previous: string;
  results: T;
}
export interface Links {
  next: string | null;
  previous: string | null;
}
export type LinksMap = Map<keyof Links, Links[keyof Links]>;

export interface FetchOptions {
  limit?: number;
  offset?: number;
  link?: string;
  search?: string;
  id?: number;
}
