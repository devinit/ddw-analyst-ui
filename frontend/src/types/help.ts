export type MenuLink = {
  caption: string;
  url: string;
  disabled?: boolean;
  addDividerAfter?: boolean;
};
