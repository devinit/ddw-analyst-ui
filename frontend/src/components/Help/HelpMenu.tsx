import React, { Fragment, FunctionComponent } from 'react';
import { Dropdown, NavLink } from 'react-bootstrap';
import styled from 'styled-components';
import { MenuLink } from '../../types/help';

const StyledDropdownItem = styled(Dropdown.Item)`
  text-transform: capitalize !important;

  &.disabled {
    opacity: 0.6;
  }
  &:hover,
  &:active {
    color: #ffffff !important;
  }
`;

type HelpMenuProps = {
  links: MenuLink[];
};

const HelpMenu: FunctionComponent<HelpMenuProps> = ({ links }) => {
  return (
    <>
      {links.map((link) => (
        <Fragment key={link.caption.split(' ').join('').toLowerCase()}>
          <StyledDropdownItem as={NavLink} href={link.url} target="_blank" disabled={link.disabled}>
            {link.caption}
          </StyledDropdownItem>
          {link.addDividerAfter ? <Dropdown.Divider /> : null}
        </Fragment>
      ))}
    </>
  );
};

export { HelpMenu };
