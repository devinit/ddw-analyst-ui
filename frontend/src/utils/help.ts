import { MenuLink } from '../types/help';

const BASE_URL = 'https://docs.google.com/document/d/'; // eslint-disable-line @typescript-eslint/naming-convention
export const myDatasetsHelpMenuLinks: MenuLink[] = [
  {
    caption: 'About Page',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.mot6r1tf2oag`,
    addDividerAfter: true,
  },
  {
    caption: 'What you should see',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.aixtc6smfj18`,
  },
  {
    caption: 'Search Datasets',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.8ttx5zp1q4fw`,
  },
  {
    caption: 'Filter by data source',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.8hgi5rikj8mx`,
  },
  {
    caption: 'Edit Dataset',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.blk96jwarais`,
  },
  {
    caption: 'View Dataset Data',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.pu36onslnft8`,
  },
  {
    caption: 'SQL Query',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.fz7bwxbta45l`,
  },
  {
    caption: 'Export to CSV',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.hceq6r48n1ir`,
  },
  {
    caption: 'Versions',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.cyp9venaoova`,
    addDividerAfter: true,
  },
  {
    caption: 'Report Issue',
    url: `https://github.com/devinit/ddw-analyst-ui/issues/new`,
  },
];
export const publishedDatasetsHelpMenuLinks: MenuLink[] = [
  {
    caption: 'About Page',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.id1y6txotno6`,
    addDividerAfter: true,
  },
  {
    caption: 'What you should see',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.nht4254tao20`,
  },
  {
    caption: 'Search Datasets',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.o7uiocmzh6fs`,
  },
  {
    caption: 'Filter by data source',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.28x8odk5fbr7`,
  },
  {
    caption: 'Action Buttons',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.gt23wxdp2igl`,
    addDividerAfter: true,
  },
  {
    caption: 'Report Issue',
    url: `https://github.com/devinit/ddw-analyst-ui/issues/new`,
  },
];
export const dataSourcesHelpMenuLinks: MenuLink[] = [
  {
    caption: 'About Page',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.s33dnwookv04`,
    addDividerAfter: true,
  },
  {
    caption: 'What you should see',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.shazlauczcxq`,
  },
  {
    caption: 'Metadata',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.wjys5kg5a8op`,
  },
  {
    caption: 'Datasets',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.pipbmwii2f31`,
  },
  {
    caption: 'Versions',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.aqgsqx8tl2qp`,
    addDividerAfter: true,
  },
  {
    caption: 'Report Issue',
    url: `https://github.com/devinit/ddw-analyst-ui/issues/new`,
  },
];
export const queryBuilderHelpMenuLinks: MenuLink[] = [
  {
    caption: 'About Page',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.f7o0k8sboyg6`,
    addDividerAfter: true,
  },
  {
    caption: 'Create/Edit Dataset',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.tfpa8t26iluz`,
  },
  {
    caption: 'Add Step Button',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.4vspcej3998n`,
  },
  {
    caption: 'Select Query',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.bovg272i2us6`,
  },
  {
    caption: 'Filter Query',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.rz9bz6cinblo`,
  },
  {
    caption: 'Join Query',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.vafsz1yr8ewa`,
  },
  {
    caption: 'Aggregate Query',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.r7eozdejatca`,
    addDividerAfter: true,
  },
  {
    caption: 'Report Issue',
    url: `https://github.com/devinit/ddw-analyst-ui/issues/new`,
  },
];
export const queryHistoryHelpMenuLinks: MenuLink[] = [
  {
    caption: 'About Page',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.cyp9venaoova`,
    addDividerAfter: true,
  },
  {
    caption: 'Report Issue',
    url: `https://github.com/devinit/ddw-analyst-ui/issues/new`,
  },
];
export const updateDataSourceHelpMenuLinks: MenuLink[] = [
  {
    caption: 'About Page',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.upwq5038usht`,
    addDividerAfter: true,
  },
  {
    caption: 'Select Data Source',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.1d7pqkh1xuui`,
  },
  {
    caption: 'Select CSV',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.hrv49fqln9rm`,
  },
  {
    caption: 'Map',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.ayi38fvc433z`,
  },
  {
    caption: 'Upload',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.zdxbhxqfp23l`,
    addDividerAfter: true,
  },
  {
    caption: 'Report Issue',
    url: `https://github.com/devinit/ddw-analyst-ui/issues/new`,
  },
];
export const queryDataHelpMenuLinks = [
  {
    caption: 'About Page',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.pu36onslnft8`,
    addDividerAfter: true,
  },
  {
    caption: 'Report Issue',
    url: `https://github.com/devinit/ddw-analyst-ui/issues/new`,
  },
];
export const scheduledEventHelpMenuLinks: MenuLink[] = [
  {
    caption: 'About Page',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.j61j13xt7xdx`,
    addDividerAfter: true,
  },
  {
    caption: 'What you should see',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.il4qgaa307jk`,
  },
  {
    caption: 'View Run Instaces',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.5aho207lsxrr`,
  },
  {
    caption: 'Enable Schedule',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.2h2f51k6t5sc`,
    addDividerAfter: true,
  },
  {
    caption: 'Report Issue',
    url: `https://github.com/devinit/ddw-analyst-ui/issues/new`,
  },
];
export const sourceHistorysHelpMenuLinks: MenuLink[] = [
  {
    caption: 'About Page',
    url: `${BASE_URL}1j10m46dLY-zb5Mee4JWncTGeOsf2o8oChSN90ZUu8vk/edit#heading=h.aqgsqx8tl2qp`,
    addDividerAfter: true,
  },
  {
    caption: 'Report Issue',
    url: `https://github.com/devinit/ddw-analyst-ui/issues/new`,
  },
];
