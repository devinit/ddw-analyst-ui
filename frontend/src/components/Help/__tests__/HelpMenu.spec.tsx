/**
 * @jest-environment jsdom
 */
import 'jest-styled-components';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MenuLink } from '../../../types/help';
import { HelpMenu } from '../HelpMenu';

describe('HelpMenu', () => {
  const links: MenuLink[] = [];
  beforeEach(() => {
    links[0] = {
      caption: 'Item 1',
      url: 'https://staging-ddw.devinit.org',
    };
    links[1] = {
      caption: 'Item 2',
      url: 'https://ddw.devinit.org',
    };
  });
  afterEach(cleanup);

  test('renders correctly with the required link properties', () => {
    const renderer = TestRenderer.create(<HelpMenu links={links} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders a divider', () => {
    links[0].addDividerAfter = true;

    const { container } = render(<HelpMenu links={links} />);
    const divider = container.children[1];

    expect(divider).toHaveClass('dropdown-divider');
    expect(divider).toHaveAttribute('role', 'separator');
  });

  test('renders a disabled link', () => {
    links[1].disabled = true;

    const { container } = render(<HelpMenu links={links} />);
    const disabledLink = container.children[1];

    expect(disabledLink).toHaveClass('disabled');
  });
});
