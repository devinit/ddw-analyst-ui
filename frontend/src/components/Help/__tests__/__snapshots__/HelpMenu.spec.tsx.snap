// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpMenu renders correctly with the required link properties 1`] = `
Array [
  .c0 {
  text-transform: capitalize!important;
}

.c0.disabled {
  opacity: 0.6;
}

.c0:hover,
.c0:active {
  color: #ffffff!important;
}

<a
    className="c0 nav-link"
    href="https://staging-ddw.devinit.org"
    onClick={[Function]}
    onKeyDown={[Function]}
    target="_blank"
  >
    Item 1
  </a>,
  .c0 {
  text-transform: capitalize!important;
}

.c0.disabled {
  opacity: 0.6;
}

.c0:hover,
.c0:active {
  color: #ffffff!important;
}

<a
    className="c0 nav-link"
    href="https://ddw.devinit.org"
    onClick={[Function]}
    onKeyDown={[Function]}
    target="_blank"
  >
    Item 2
  </a>,
]
`;
